server/src/main/java/org/elasticsearch/action/ingest/PutPipelineTransportAction.java

```diff
@@ -55,6 +58,14 @@ public PutPipelineTransportAction(ThreadPool threadPool, TransportService transp
     @Override
     protected void masterOperation(Task task, PutPipelineRequest request, ClusterState state, ActionListener<AcknowledgedResponse> listener)
             throws Exception {
+        if (state.getNodes().getMinNodeVersion().before(Version.V_8_0_0)) {
+            Map<String, Object> pipelineConfig = XContentHelper.convertToMap(request.getSource(), false, request.getXContentType()).v2();
+            if (pipelineConfig.containsKey(Pipeline.META_KEY)) {
+                throw new IllegalArgumentException("All nodes in the cluster must be at " + Version.V_8_0_0 + " or higher in order to" +
+                    "create a pipeline with the " + Pipeline.META_KEY + " field. The minimum version currently in the cluster is " +
+                    state.getNodes().getMinNodeVersion());
```

Comment:
Fairly minor, but we tend towards more succinct error messages in all lowercase for checks like this one. As an example:

https://github.com/elastic/elasticsearch/blob/7.9/server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java#L126

Reply:
Done.